cfep-10, feedstock support statuses (#15)
conda-forge/cfep: cfep-10.md

<table>
<tr><td> Title </td><td> Feedstock Statuses </td>
<tr><td> Status </td><td> Proposed </td></tr>
<tr><td> Author(s) </td><td> Christopher J. "CJ" Wright &lt;cjwright4242@gmail.com&gt;</td></tr>
<tr><td> Created </td><td> Sept 29, 2019</td></tr>
<tr><td> Discussion </td><td> link to the PR where the CFEP is being discussed, NA is circulated initially </td></tr>
<tr><td> Implementation </td><td> link to the PR for the implementation, NA if not availble </td></tr>

> </table>

Reviewer: Typos: `NA is circulated initially`, `availble`.

## Abstract

This document proposes the creation of four new statuses for feedstocks that denote the	
standing of the feedstock and transistions between the standings.
These statuses will help us triage feedstocks and identify which feedstocks need

> help, maintainers, or archiving.

Reviewer: Should explicitly state that this CFEP does not "implement" these statuses in terms of saying how they should be surfaced, tracked, etc.

## Motivation

It can be difficult for maintainers and would be maintainers to identify which feedstocks could use their
help.
Additionally, it is difficult for core to know which feedstocks need additional help or archiving.
Finally, users of CF packages may find it difficult to understand which packages are dorment.

> This proposal aims to expose this information more clearly.

Reviewer: Typos: change to `would-be`, `dormant`.

> This proposal aims to expose this information more clearly.

Reviewer: The last sentence of this paragraph does not seem strictly correct. This proposal has very little to say about *exposing* status information at the moment.

## Specification

This proposal creates four new statuses

> ### In Perfect Standing

Reviewer: Nothing in this world is perfect. "Excellent standing", perhaps?

> The feedstock is actively maintained, new versions of the packages are being created as needed, autotick bot PRs are addressed 

Author: ```suggestion
The feedstock is actively maintained, new versions of the packages are being created as needed, autotick bot and manual PRs and issues are addressed 
```

Reviewer: Could just write "... created as needed, PRs and issues are addressed ..."?

> (merged, fixed, or closed) promptly. Less than 3 bot PRs are open at once.

Author: ```suggestion
(merged, fixed, or closed) promptly. Less than 3 PRs are open at once.
```

Reviewer 2: Using a hard line like this encourages maintainers close user-contributed work-in-progress PRs. Is that a goal? Bot PRs represent tasks that should be done in a timely manner, user PRs may or may not be. I've seen a number of repos with stalled "build on Windows" or "split outputs" PRs outstanding for a long time, where their incompleteness is not an indication of the repo being unmaintained.

While I think a simple numerical metric for open bot PRs is valid, it's harder to have a good metric for user PRs (just counting open doesn't seem good to me, but e.g. open and passing tests without comment since last commit for a length of time...)

Reviewer 2: I'd leave this statement as "Less than 3 bot PRs are open at once" and let community PRs be more vaguely addressed by the previous statement, relying on human judgment instead of bots to enforce "addressed promptly".

> (merged, fixed, or closed) promptly. Less than 3 bot PRs are open at once.

Reviewer: Grammar: "fewer than three PRs are open".

More maintainers are always welcome.

### In Good Standing

> Maintainers are responsive but probably overworked. Multiple bot PRs have begun to pile up.

Author: ```suggestion
Maintainers are responsive but probably overworked. Multiple PRs and issues have begun to pile up.
```

> We should advertise these as needing new maintainers.

Reviewer: For clarity, I think what is meant here is "needing *additional* maintainers", correct? "needing new maintainers" might be taken to suggest that the current ones should be kicked off, which I presume is not the intention.

### Unsupported
This feedstock has no or unresponsive maintainers and either needs new maintainers or to be archived.
Advertise that this needs maintainers and for the current ones re-engage with the feedstock or remove
themselves, otherwise the feedstock will be archived. Set a deadline of X days/weeks/months for new

> maintainers to come on board, if nothing happens the feedstock is archived.

Author: ```suggestion
maintainers to come on board or existing maintainers to re-engage, if nothing happens the feedstock is archived.
```

> maintainers to come on board or existing maintainers to re-engage, if nothing happens the feedstock is archived.

Reviewer: Note that the sentence about setting a deadline is a piece of implementation policy, not really part of the definition of the "unsupported" status.

<table>
<tr><td> Title </td><td> Feedstock Statuses </td>

> <tr><td> Status </td><td> Proposed </td></tr>

Reviewer 3: ```suggestion
<tr><td> Status </td><td> Deferred </td></tr>
```
